### PR TITLE
add core code of valle

### PR DESCRIPTION
**Pull request by lmxue**
[Vall-E](https://arxiv.org/abs/2301.02111) is a zero-shot TTS architecture that uses a neural codec language model with discrete codes. This PR is to support Vall-E in Amphion.
* bins/tts
* config/valle.json
* egs/tts/VALLE
* models/tts/valle

### CODE REVIEW

File: processors/content_extractor.py [hunk 1]
```diff
@@ -24,7 +24,9 @@
 from modules import whisper_extractor as whisper
 from modules.wenet_extractor.utils.init_model import init_model
 from modules.wenet_extractor.utils.checkpoint import load_checkpoint
-
+from utils.tokenizer import G2PModule, tokenize_text
+from utils.symbol_table import SymbolTable
+from text.g2p import preprocess_english, read_lexicon
 
 """
     Extractor for content features
```
Comment (RMSnow) [inline processors/content_extractor.py:32]: Provide the comments of `extract_phoneme` related code

File: processors/content_extractor.py [hunk 2]
```diff
@@ -539,3 +541,49 @@ def extract_utt_content_features_dataloader(cfg, metadata, num_workers):
                 )
                 for index, utt in enumerate(_metadata):
                     extractor.save_feature(utt, batch_content_features[index])
+
+    if cfg.preprocess.extract_phoneme:
```
Comment (RMSnow) [inline processors/content_extractor.py:545]: The current code will make SVC and TTS entangle unnecessarily. Move Line545-589 to a new function.

**Review by RMSnow — APPROVED** [review]
Leave the comments as the next improvement

File: config/base.json [hunk 3]
```diff
@@ -177,7 +181,6 @@
     "valid_interval": 10000,
     "keep_checkpoint_max": 5,
     "multi_speaker_training": false, // True: train multi-speaker model; False: training single-speaker model;
-    "gradient_accumulation_step": 1,
```
Comment (RMSnow) [inline config/base.json:180]: why delete?

File: config/base.json
Comment (RMSnow) [inline config/base.json:1]: Move all the configs about TTS into a TTS's config base json

File: bins/tts/inference.py [hunk 4]
```diff
@@ -139,13 +139,12 @@ def build_parser():
...
 
-
 def main():
     # Parse arguments
-    args = build_parser().parse_args()
-    # args, infer_type = formulate_parser(args)
-    print("args: ", args)
-
+    parser = build_parser()
+    VALLEInference.add_arguments(parser)
```
Comment (zhizhengwu) [inline bins/tts/inference.py:145]: this looks like VALLEInference is used no matter what type of the model.

File: bins/tts/preprocess.py [hunk 5]
```diff
@@ -53,10 +61,20 @@ def extract_content_features(dataset, output_path, cfg, num_workers=1):
...
-    types = ["train", "test"] if "eval" not in dataset else ["test"]
+    # types = ["train", "test"] if "eval" not in dataset else ["test"]
+
+    types = list()
+    types.append((cfg.preprocess.train_file).split('.')[0])
+    types.append((cfg.preprocess.valid_file).split('.')[0])
+    if 'test' not in types: 
+        types.append('test') 
+    if "eval" in dataset:
+        types = ["test"]
```
Comment (zhizhengwu) [inline bins/tts/preprocess.py:72]: repeating lines: 32 - 39

File: bins/tts/preprocess.py [hunk 6]
```diff
@@ -53,10 +61,20 @@ def extract_content_features(dataset, output_path, cfg, num_workers=1):
...
+            
     metadata = []
     for dataset_type in types:
         dataset_output = os.path.join(output_path, dataset)
+        # dataset_file = os.path.join(dataset_output, "{}.json".format(dataset_type))
         dataset_file = os.path.join(dataset_output, "{}.json".format(dataset_type))
```
Comment (zhizhengwu) [inline bins/tts/preprocess.py:78]: duplicating line 78?

File: bins/tts/train.py [hunk 7]
```diff
@@ -77,12 +93,13 @@ def main():
             new_datasets_list.extend(filter(None, new_datasets))
         cfg.dataset.extend(new_datasets_list)
 
-    # CUDA settings
+    # # CUDA settings
```
Comment (zhizhengwu) [inline bins/tts/train.py:96]: No need to add one more '#'

**Comment by zhizhengwu** [comment]
We should provide demos/samples in a PR

File: processors/acoustic_extractor.py [hunk 8]
```diff
@@ -211,6 +212,11 @@ def __extract_utt_acoustic_features(dataset_output, cfg, utt):
             label = audio_to_label(wav, cfg.preprocess.bits)
             save_feature(dataset_output, cfg.preprocess.label_dir, uid, label)
 
+        if cfg.preprocess.extract_acoustic_token:
```
Comment (RMSnow) [inline processors/acoustic_extractor.py:215]: Do not modify `__extract_utt_acoustic_features` anymore. It is not a common extraction pipeline now. See `extract_utt_acoustic_features_tts` (line221) and `extract_utt_acoustic_features_vocoder`(line233) as reference. Please move all the functions of TTS's acoustic feature extraction into line221.

**Review by RMSnow — CHANGES_REQUESTED** [review]

File: bins/tts/inference.py [hunk 9]
```diff
@@ -75,9 +73,9 @@ def build_parser():
     )
     parser.add_argument(
         "--text",
-        help="Text to be synthesized",
+        help="Text",
```
Comment (zhizhengwu) [inline bins/tts/inference.py:76]: 'Text to be synthesized' is more informative than 'Text'

**Review by RMSnow — APPROVED** [review]

**Pull request merged**